src/core/highlight.js

```diff
@@ -45,32 +45,44 @@ export async function run(conf) {
         console.error("Timed-out waiting for highlight:", element);
         done();
       }, 4000);
-      const msg = {
-        action: "highlight",
-        code: element.textContent,
-        id: `highlight:${i}`,
-        languages: getLanguageHint(element.classList),
-      };
-      worker.addEventListener("message", function listener(ev) {
-        const {
-          data: { id, language, value },
-        } = ev;
-        if (id !== msg.id) {
-          return; // not for us!
-        }
-        element.innerHTML = value;
-        if (element.localName === "pre") {
-          element.classList.add("hljs");
-        }
-        if (language) {
-          element.classList.add(language);
-        }
-        clearTimeout(timeoutId);
-        worker.removeEventListener("message", listener);
-        done();
+      const elementsToHighlight =
+        element.querySelectorAll("code").length > 0
+          ? Array.from(element.querySelectorAll("code"))
+          : [element];
+      elementsToHighlight.map((each, id) => {
```

Comment:
using this id is problematic, because if you have multiple `pre`s with multiple `code`s you are still going get messages to the wrong element. You need to either:

1. combine the outer (`pre`) id with this `code` `id`, 
1. or have a module level message `id` that gets incremented. 
1. or repurpose the `timeoutId` as the message id. Those are also unique.

Reply:
Ah, oh yeah! :o

Reply:
instead of using a global `id` , we could pass a pair `(i, id)` won't require much modification .

Reply:
The combination id is fine, so long as putting the id together is not more computationally expensive than doing i++.